server/src/main/java/org/elasticsearch/reservedstate/service/ReservedStateVersion.java

```diff
@@ -32,7 +32,7 @@ public record ReservedStateVersion(Long version, Version compatibleWith) impleme
         "reserved_cluster_state_version_metadata",
         a -> {
             Long updateId = Long.parseLong((String) a[0]);
-            Version minCompatVersion = Version.fromString((String) a[1]);
+            BuildVersion minCompatVersion = BuildVersion.fromString((String) a[1]);
```

Comment:
This is ok for serverless, as whatever the value is will just be ignored and the singleton used